### PR TITLE
BUGFIX: SD-867 only support geo/location DF on Location bucket

### PR DESCRIPTION
**Pull request by naibinh**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2870
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2608

# Related Jira tasks
- SD-867: https://jira.intgdc.com/browse/SD-867

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6460/ : SUCCESS in 7m 31s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2251/ : SUCCESS in 5s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6461/ : SUCCESS in 9m 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2252/ : SUCCESS in 9s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3351/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2253/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2758/ : SUCCESS in 15m 34s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1433.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2254/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2759/ : SUCCESS in 23m 50s

File: src/internal/constants/uiConfig.ts [hunk 1]
```diff
@@ -574,6 +574,7 @@ export const GEO_PUSHPIN_CHART_UICONFIG: IUiConfig = {
     buckets: {
         location: {
             ...geoAttributesBase,
+            acceptLocationDisplayForm: true,
```
Comment (ptquang86) [inline src/internal/constants/uiConfig.ts]: shoud it be `accepts...` ?

File: src/internal/interfaces/Visualization.ts [hunk 2]
```diff
@@ -209,6 +209,7 @@ export interface IBucketUiConfig {
     allowShowOnSecondaryAxis?: boolean;
     allowSelectChartType?: boolean;
     allowOptionalStacking?: boolean;
+    acceptLocationDisplayForm?: boolean;
```
Comment (ptquang86) [inline src/internal/interfaces/Visualization.ts]: We have some conflicting name here `allow...` vs `allows...`
https://github.com/gooddata/gooddata-react-components/blob/f0e7446990718924a2e4ce0b692c20ac33292a31/src/internal/interfaces/Visualization.ts#L201

In this case, I prefer `accepts...`.
What do you think?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6477/ : SUCCESS in 16m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2267/ : SUCCESS in 6s

File: src/internal/interfaces/Visualization.ts (hunk 2 above)
Comment (naibinh) [reply src/internal/interfaces/Visualization.ts]: I would go with `accepts` as well. Thanks

File: src/internal/constants/uiConfig.ts (hunk 1 above)
Comment (naibinh) [reply src/internal/constants/uiConfig.ts]: Fixed

**Comment by ptquang86** [comment]
LGTM

**Comment by lvkhanh** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6481/ : SUCCESS in 7m 27s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2276/ : SUCCESS in 4s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6504/ : SUCCESS in 23m 49s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2303/ : SUCCESS in 8s

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3387/ : SUCCESS in 14m 49s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6528/ : SUCCESS in 17m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1573/ : SUCCESS in 25s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2262/ : SUCCESS in 23m 54s

**Pull request merged**